osu.Game/Screens/SelectV2/BeatmapCarousel.cs

```diff
@@ -252,6 +261,32 @@ private void setExpansionStateOfSetItems(BeatmapSetInfo set, bool expanded)
 
         #endregion
 
+        #region Animation
+
+        /// <summary>
+        /// Moves non-selected beatmaps to the right, hiding off-screen.
+        /// </summary>
+        public bool VisuallyFocusSelected { get; set; }
+
+        private float selectionFocusOffset;
+
+        protected override void Update()
+        {
+            base.Update();
+
+            selectionFocusOffset = (float)Interpolation.DampContinuously(selectionFocusOffset, VisuallyFocusSelected ? 300 : 0, 100, Time.Elapsed);
+
+            foreach (var panel in Scroll.Panels)
+            {
+                var c = (ICarouselPanel)panel;
+
+                if (!c.Selected.Value)
+                    panel.X += selectionFocusOffset;
```

Comment:
I dunno if this is just me but to me this looked turbo sketchy as to how this could ever work until I realised that the base call was changing `panel.X` every frame which is why this works. Not sure if anything is actionable here, just remarking I guess

Reply:
Pushed an alternative, see what you think.

Reply:
Yeah sure, works for me.